openpype/client/entities.py

```diff
@@ -1033,6 +1080,31 @@ def get_representation_by_name(
     return conn.find_one(query_filter, _prepare_fields(fields))
 
 
+def get_representation_by_task(
+    project_name: str, task_name: str, version_doc: dict
+) -> str:
+    """Return representation for given project and task, and version.
+
+    Args:
+        project_name (str): Project name.
+        task_name (str): Task name.
+        version_doc (dict): Version doc.
+
+    Returns:
+        str: Representation.
+    """
+    return next(
+        (
+            representation
+            for representation in get_representations(
+                project_name, version_ids=[version_doc["_id"]]
+            )
+            if representation["context"]["task"]["name"] == task_name
```

Comment:
There are representations without `"task"` in context.

Reply:
I have similar issue with this function as with `match_subset_id`. It make sense only in very specific use-cases and since it is in fact just a filtering function I don't see a benefit to have it in public api for entities.

Reply:
This comes from #4102 I'll look into it though, but as part of a fix for #4102

Reply:
I see. Almost none of the comments here are not related to this PR at all...